cfpb/cfgov-refresh
# Optimize FWB JS and fix an IE error

The script had trouble keeping the state of the form correctly saved
when returning to the page after submission. To better track each
individual question the script sets up an object and tracks the radio states in
relation to their parent fieldset. In addition I made some optimizations
and broke code up into smaller testable bites where possible.

## Changes

- Added onload checker to fix issue with IE not immediately recognizing inputs as checked when returning to the page after submission
- Modularized the functionality as much as possible to make testing easier
- Grouped radios by parent fieldset to simplify completion checks

## Testing

1. Open `10.0.2.2:8000/consumer-tools/financial-well-being/` in IE 9
2. Verify the submit button is disabled
3. Randomly select a radio from each fieldset group (I just use the left hand one for all of them)
4. Submit
5. Return to page using browser back button, ensure your choices are still selected and the submit button is not disabled.
6. run `gulp test` and ensure everything passes.

## Todos

- ~~We still need to have a discussion around satellite apps in cfgov-refresh.~~
- ~~Needs unit tests (wanted to get some feedback first).~~
- Could probably stand to use promises instead of one function triggering the next.
- ~~ES6-ify all the things.~~

## Checklist
